modules/player/api_test.go

```diff
@@ -136,44 +136,76 @@ func TestModuleHandleMovementEvent(t *testing.T) {
 	})
 }
 
+func withinError(x, y float64, diff float64) bool {
+	if x+diff > y && x-diff < y {
+		return true
+	}
+	return false
+}
+
+func withinErrorVec3(a, b mgl.Vec3, diff float64) bool {
+	if withinError(a.X(), b.X(), diff) && withinError(a.Y(), b.Y(), diff) &&
+		withinError(a.Z(), b.Z(), diff) {
+		return true
+	}
+	return false
+}
+
+func withinErrorQuat(q1 mgl.Quat, q2 mgl.Quat, diff float64) bool {
+	if withinError(q1.W, q2.W, diff) && withinErrorVec3(q1.V, q2.V, diff) {
+		return true
+	}
+	return false
+}
+
+const errMargin = 0.000001
+
 func TestModuleHandleLookEvent(t *testing.T) {
 	t.Parallel()
 
 	testCases := []struct {
-		right    float32
-		down     float32
-		rotation glm.Quat
+		right    float64
+		down     float64
+		rotation mgl.Quat
 	}{
 		{
 			right: 1.0,
 			down:  0.0,
-			rotation: glm.Quat{
+			rotation: mgl.Quat{
 				W: math.Cos(1.0 / 2),
-				V: [3]float32{0, -math.Sin(1.0 / 2), 0},
+				V: [3]float64{0, -math.Sin(1.0 / 2), 0},
 			},
 		},
 		{
 			right: -1.0,
 			down:  0.0,
-			rotation: glm.Quat{
+			rotation: mgl.Quat{
 				W: math.Cos(1.0 / 2),
-				V: [3]float32{0, math.Sin(1.0 / 2), 0},
+				V: [3]float64{0, math.Sin(1.0 / 2), 0},
 			},
 		},
 		{
 			right: 0.0,
 			down:  1.0,
-			rotation: glm.Quat{
+			rotation: mgl.Quat{
 				W: math.Cos(1.0 / 2),
-				V: [3]float32{-math.Sin(1.0 / 2), 0, 0},
+				V: [3]float64{-math.Sin(1.0 / 2), 0, 0},
 			},
 		},
 		{
 			right: 0.0,
 			down:  -1.0,
-			rotation: glm.Quat{
+			rotation: mgl.Quat{
 				W: math.Cos(1.0 / 2),
-				V: [3]float32{math.Sin(1.0 / 2), 0, 0},
+				V: [3]float64{math.Sin(1.0 / 2), 0, 0},
+			},
+		},
+		{
+			right: math.Pi / 2.0,
+			down:  math.Pi / 2.0,
+			rotation: mgl.Quat{
+				W: 1.0 / 2.0,
+				V: [3]float64{1.0 / 2.0, -1.0 / 2.0, -1.0 / 2.0},
```

Comment:
I don't think this test case is correct. That rotation gives a result of `{1, 0, 0}`. See [wolframalpha](https://www.wolframalpha.com/input/?i=quaternion+%281%2F2%2C1%2F2%2C-1%2F2%2C-1%2F2%29+*+quaternion+%280%2C+0%2C+0%2C+-1%29+*+quaternion+%281%2F2%2C+-1%2F2%2C+1%2F2%2C+1%2F2%29).

Reply:
Doing just the X rotation also gives `{1,0,0}`. See [wolframalpha](https://www.wolframalpha.com/input/?i=quaternion+%28sqrt%5B2%5D%2F2%2C0%2C-sqrt%5B2%5D%2F2%2C0%29+*+quaternion+%280%2C+0%2C+0%2C+-1%29+*+quaternion+%28sqrt%5B2%5D%2F2%2C+0%2C+sqrt%5B2%5D%2F2%2C+0%29). Clearly, our math is wrong in the Y direction.

Reply:
[This ](https://www.wolframalpha.com/input/?i=quaternion+%281%2F2%2C-1%2F2%2C-1%2F2%2C-1%2F2%29+*+quaternion+%280%2C+0%2C+0%2C+-1%29+*+quaternion+%281%2F2%2C+1%2F2%2C+1%2F2%2C+1%2F2%29) is the result we want, which I computed without rotating the Y direction axis of rotation.

Reply:
OK I understand. The result you give rotates the up vector from +y to +x, which is what we want.